### ISSUE TITLE
1.18.0 contains breaking changes

### ISSUE THREAD

**Issue by manniniandrea** [browserstack/browserstack-cypress-cli#398]
We use Nx to build and test our code on our pipeline in parallel, since we have more than one target running at the same time we must differentiate each cypress config file name.
For example, when we run our cypress tests these are the config file generated:
`cypress.target-1.json`
`cypress.target-2.json`
`cypress.target-3.json`

Inspecting the code we've found out that the new version of the library only allows standard names such as `cypress.json`, `cypress.config.js`, etc.

As you can see in the `verify` function here 
https://github.com/browserstack/browserstack-cypress-cli/blame/566e0f43db9a190ce978a65236324ba98708a747/bin/helpers/capabilityHelper.js#L252
and here
https://github.com/browserstack/browserstack-cypress-cli/blame/566e0f43db9a190ce978a65236324ba98708a747/bin/helpers/constants.js#L391

every configuration file named differently from the below ones will get discarded

- cypress.json
- cypress.config.js
- cypress.config.ts
- cypress.config.mjs
- cypress.config.cjs

Can the ability to ingest files with custom names be reintroduced in the latest version?

Thank you.

**Comment by SachinMadhavan** [comment]
We would like to share that there’s no difference between v1.17.0 and v1.18.0 in terms of features. CLI version v1.18.0 was introduced with the same features of v1.17.0 with an additional support for Cypress v10. Hence, you do not need upgrade to v1.18.0 unless you want to use Cypress v10. 
​
Additionally, the cypress_config_file key’s primary application is to pass the cypress.json file from different paths, but we were not aware that customers were also using this key to pass different file names other than cypress.json. 

When we do the next CLI release, we will add this support for Cypress 9 and below. For Cypress 10, we understand it is a limitation, but we will need to explore how to implement this since the structure is different.

**Comment by SachinMNair** [comment]
Hey @manniniandrea I would like to inform you that our team has released a new version of BrowserStack Cypress CLI: https://github.com/browserstack/browserstack-cypress-cli/releases (v1.19.0). Do try it out and let us know in case you face any issue.

**Comment by SachinMadhavan** [comment]
Hey @manniniandrea I would like to inform you that a new version of BrowserStack Cypress CLI: https://github.com/browserstack/browserstack-cypress-cli/releases (v1.19.0) has been released. Do try it out and let us know in case you face any issue.

**Comment by jjmluz** [comment]
Hi @SachinMadhavan,
We were facing the same issue as @manniniandrea with **v1.18.0** so we renamed our `cypress-e2e.config.ts` file to `cypress.config.js` to be able to run the tests.

After upgrading to **v1.19.0**, we are now able to name it `cypress-e2e.config.js` but still can't name it `cypress-e2e.config.ts`. We get the error `Your configFile is invalid` on Automate Dashboard. I have also tried with `cypress.config.ts` and got the same error so the issue seems to be with the `.ts` extension.

Can `.ts` files be supported on the next release? Thanks!

**Comment by jjmluz** [comment]
Actually, you may disregard my comment above. The issue had nothing to do with the config file name.
We were able to overcome the error on Automate Dashboard by adding `typescript` to the `npm_dependencies` on `browserstack.json` file.

TLDR: `cypress-e2e.config.ts` works now! Thanks!

**Comment by manniniandrea** [comment]
> Hey @manniniandrea I would like to inform you that a new version of BrowserStack Cypress CLI: https://github.com/browserstack/browserstack-cypress-cli/releases (v1.19.0) has been released. Do try it out and let us know in case you face any issue.

I can confirm that the new version definitely fixes our issues. Thank you very much.
